src/main/java/codeu/model/data/Hashtag.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.model.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/** Class representing a HSashtag. */
+public class Hashtag {
+
+  private final UUID id;
+  private final String content;
+  private List<String> userSource;
+  private List<String> conversationSource;
+  private final Instant creation;
+
+  /** Constructs a new Hashtag. */
+  public Hashtag(String content) {
+    this.id = UUID.randomUUID();
+    this.content = content.toLowerCase();
+    this.creation = Instant.now();
+    this.userSource = new ArrayList<String>();
+    this.conversationSource = new ArrayList<String>();
+  }
+
+  /** Returns the ID of this Hashtag. */
+  public UUID getId() {
+    return this.id;
+  }
+
+  /** Returns the creation time of this Hashtag. */
+  public Instant getCreationTime() {
+    return this.creation;
+  }
+
+  /** Returns the content of this Hashtag. */
+  public String getContent() {
+    return this.content;
+  }
+
+  /** Returns the String representation of the users that contain this Hashtag. */
+  public String getUserSource() {
+    return String.join(",", this.userSource);
+  }
+
+  /** Returns the String representation of the conversations that contain this Hashtag. */
+  public String getConversationSource() {
+    return String.join(",", this.conversationSource);
+  }
+
+  /**
+   * Add the ID of the user to the userSource. Returns true if the String representation of the User
+   * is added; returns false if the String representation of the User exists.
+   */
+  public boolean addUser(UUID userID) {
```

Comment:
Why is it necessary to have this function as a boolean (instead of just void) ? it is just supposed to add or not add whenever necessary. Are you planning on using this boolean somewhere else?

Reply:
Yeah, Donnell's comment is what I am trying to do. If the tag already has this user in the list, the addUser will return false as a indicator that this user already exists(this user might have the same hashtag occur more than once in his/her readme). It's true that what it returns does not make a difference at this stage, what I do think there is a difference in actions.

Reply:
oh ok. that makes sense!